### PR TITLE
Save data async to prevent lag spikes

### PR DESCRIPTION
**Pull request by booky10**
People with slow disks will experience lag spikes when saving on death.
This pull request creates a separate thread in which each load and save is executed.
The load method returns a CompletableFuture so that the caller can wait until the entries are fully loaded into the list.

### CODE REVIEW

File: src/main/java/com/glisco/deathlog/storage/BaseDeathLogStorage.java [hunk 1]
```diff
@@ -57,6 +71,11 @@ protected void load(File file, List<DeathInfo> into) {
     }
 
     protected void save(File file, List<DeathInfo> list) {
+        // Copying the list because of concurrency issues
+        FS_EXECUTOR.execute(() -> save0(file, ImmutableList.copyOf(list)));
```
Comment (BasiqueEvangelist) [inline src/main/java/com/glisco/deathlog/storage/BaseDeathLogStorage.java]: Why are you copying it in the lambda and not in the outer method?

Comment (booky10) [reply src/main/java/com/glisco/deathlog/storage/BaseDeathLogStorage.java]: Smaller diff + you need less indents in the save0 method.

Comment (BasiqueEvangelist) [reply src/main/java/com/glisco/deathlog/storage/BaseDeathLogStorage.java]: Well yes, but then you're going to copy the list in the thread pool, which can cause concurrency issues

Comment (booky10) [reply src/main/java/com/glisco/deathlog/storage/BaseDeathLogStorage.java]: Oh, sorry, I misunderstood your question, will fix it.

**Comment by gliscowo** [comment]
Thanks for this PR and pointing out the problem. I've recreated what you've done here in a way that I feel suits the codebase a bit better in dc8c940, so I'll go ahead and close this now
Cheers

**Pull request closed without merging**